uitools/cpp/Esri/ArcGISRuntime/Toolkit/FloorFilterController.cpp

```diff
@@ -0,0 +1,940 @@
+
+    if (m_settingViewpoint)
+      return;
+
+    Viewpoint observedViewpoint;
+    if (auto sceneView = qobject_cast<SceneViewToolkit*>(m_geoView))
+    {
+      observedViewpoint = sceneView->currentViewpoint(ViewpointType::CenterAndScale);
+    }
+    else if (auto mapView = qobject_cast<MapViewToolkit*>(m_geoView))
+    {
+      observedViewpoint = mapView->currentViewpoint(ViewpointType::CenterAndScale);
+    }
+
+    // Expectation: viewpoint is center and scale
+    if (observedViewpoint.isEmpty() ||
+        isnan(observedViewpoint.targetScale()))
+      return;
+
+    auto floorManager = getFloorManager(m_geoView);
+
+    // Only take action if viewpoint is within minimum scale. Default minscale is 4300 or less (~zoom level 17 or greater)
+    double targetScale = 0.0;
+    if (floorManager)
+    {
+      if (auto siteLayer = floorManager->siteLayer())
+      {
+        targetScale = siteLayer->minScale();
+      }
+    }
+
+    if (targetScale == 0.0)
+      targetScale = 4300.0;
+
+    if (observedViewpoint.targetScale() > targetScale)
+    {
+      if (m_automaticSelectionMode == AutomaticSelectionMode::Always)
+      {
+        setSelectedLevelId({});
+        setSelectedFacilityId({});
+        setSelectedSiteId({});
+      }
+      // Assumption: if too zoomed out to see sites, also too zoomed out to see facilities
+      return;
+    }
+
+    // If the centerpoint is within a site's geometry, select that site.
+    // This code gracefully skips selection if there are no sites or no matching sites
+    auto sites = floorManager ? floorManager->sites() : QList<FloorSite*>{};
+    auto siteResult = std::find_if(std::cbegin(sites), std::cend(sites),
+                                   [&observedViewpoint](FloorSite* site)
+                                   {
+                                     if (site == nullptr)
+                                       return false;
+
+                                     auto extent = site->geometry().extent();
+                                     if (extent.isEmpty())
+                                       return false;
+
+                                     return GeometryEngine::intersects(extent, observedViewpoint.targetGeometry());
+                                   });
+
+    if (siteResult != std::cend(sites))
+    {
+      setSelectedSiteId((*siteResult)->siteId());
+    }
+    else if (m_automaticSelectionMode == AutomaticSelectionMode::Always)
+    {
+      setSelectedSiteId({});
+    }
+
+    // Move on to facility selection. Default to map-authored Facility MinScale. If MinScale not specified or is 0, default to 1500.
+    targetScale = 0.0;
+    if (floorManager)
+    {
+      if (auto siteLayer = floorManager->facilityLayer())
+      {
+        targetScale = siteLayer->minScale();
+      }
+    }
+
+    if (targetScale == 0)
+    {
+      targetScale = 1500;
+    }
+
+    // If out of scale, stop here
+    if (observedViewpoint.targetScale() > targetScale)
+    {
```

Comment:
Just an observation that there is a bit of a mixture of using brackets around single line ifs in this code. Good to settle one one style maybe